### PR TITLE
Use reverse-resolved DNS info in the connections table.

### LINKED ISSUE #1121: Use reverse DNS info
We could
- name pseudo nodes based on it
- use it to identify well know services and break them out of the internet node (#1082)
- in the connections details panel for the internet nodes (see #1017)

### PR DESCRIPTION
**Pull request by tomwilkie**
Fixes #1121 

screen shot 2016-04-20 at 17 12 21

<s>@davkal / @foot I added the `truncate` class to the details table cells, but I don't see to get tooltips.  Mind having a look?</s>

### CODE REVIEW

File: render/detailed/connections.go [hunk 1]
```diff
@@ -206,10 +211,15 @@ func connectionRows(r report.Report, in map[connection]int, includeLocal bool) [
 			connection.Linkable = false
 		}
 		if includeLocal {
+			// Does localNode have a DNS record in it?
+			label := row.localAddr
+			if set, ok := row.localNode.Sets.Lookup(endpoint.ReverseDNSNames); ok && len(set) > 0 {
+				label = set[0]
+			}
```
Comment (paulbellamy) [inline render/detailed/connections.go]: Just picking the first one seems a bit random. any better heuristics we could use? Shortest? Longest?

File: render/detailed/connections.go [hunk 2]
```diff
@@ -144,6 +147,8 @@ func outgoingConnectionsSummary(topologyID string, r report.Report, n report.Nod
 					port:       port,
 				}
 				if isInternetNode(n) {
+					endpointNode := r.Endpoint.Nodes[remoteEndpointID]
+					key.localNode = &endpointNode
```
Comment (paulbellamy) [inline render/detailed/connections.go]: let's inline these

Comment (tomwilkie) [inline render/detailed/connections.go]: Can't take a pointer to a map entry unfortunately.

File: render/detailed/connections.go (hunk 1 above)
Comment (tomwilkie) [inline render/detailed/connections.go]: This is what we used to do.

I was tempter to instead do some string formatting, so it came out as "<hostname> (<ipadd>)".  WDYT?

Comment (paulbellamy) [inline render/detailed/connections.go]: Could be quite long, but worth a shot.

File: render/detailed/connections.go (hunk 2 above)
Comment (paulbellamy) [inline render/detailed/connections.go]: eugh

**Comment by tomwilkie** [comment]
PTAL @paulbellamy

**Pull request merged**